optuna/study/study.py

```diff
@@ -45,6 +47,26 @@
 _logger = logging.get_logger(__name__)
 
 
+def _convert_positional_args(n_positional_args: int = 0) -> Any:
+    def decorator(func: Any) -> Any:
+        @wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            assert len(args) <= n_positional_args, "Too many positional arguments."
+            if len(args) >= 1:
+                warnings.warn(
+                    f"{func.__name__}: Positional arguments are deprecated."
+                    " Please give all values as keyword arguments."
+                )
+            for val, arg_name in zip(args, list(signature(func).parameters)):
```

Comment:
It might be a good idea to leave a code comment (somewhere in `_convert_positional_args` ?) for maintainers that the order of arguments (although keyword-only) may not be changed, at the decorated functions.

Reply:
Respond in 2ce7150 (add comment) and fdcf2e0 (raise TypeError), f8d64b9 (arg_names as input)